apps/bmd/src/AppRoot.tsx

```diff
@@ -141,898 +147,1091 @@ export const activationStorageKey = 'activation'
-    isVoterCardValid: true,
-    pauseProcessingUntilNoCardPresent: false,
-    voterCardCreatedAt: 0,
-  }
-
-  private initialVoterState: UserState = {
-    ballotCreatedAt: 0,
-    ballotStyleId: '',
-    contests: [],
-    precinctId: '',
-    userSettings: { textSize: GLOBALS.TEXT_SIZE },
-    votes: blankBallotVotes,
-  }
+const initialCardState: CardState = {
+  isAdminCardPresent: false,
+  isPollWorkerCardPresent: false,
+  isRecentVoterPrint: false,
+  isVoterCardExpired: false,
+  isVoterCardVoided: false,
+  isVoterCardPresent: false,
+  isVoterCardPrinted: false,
+  isVoterCardValid: true,
+  pauseProcessingUntilNoCardPresent: false,
+  voterCardCreatedAt: 0,
+}
 
-  private initialHardwareState: HardwareState = {
-    hasAccessibleControllerAttached: false,
-    hasCardReaderAttached: true,
-    hasChargerAttached: true,
-    hasLowBattery: false,
-    hasPrinterAttached: true,
-    machineConfig: { appMode: VxMarkOnly, machineId: '0000' },
-  }
+const initialVoterState: UserState = {
+  ballotCreatedAt: 0,
+  ballotStyleId: '',
+  contests: [],
+  precinctId: '',
+  shortValue: '{}',
+  userSettings: { textSize: GLOBALS.TEXT_SIZE },
+  votes: undefined,
+}
 
-  private sharedState: SharedState = {
-    appPrecinctId: '',
-    ballotsPrintedCount: 0,
-    electionDefinition: undefined,
-    isLiveMode: false,
-    isPollsOpen: false,
-    shortValue: '{}',
-    tally: [],
-  }
+const initialHardwareState: HardwareState = {
+  hasAccessibleControllerAttached: false,
+  hasCardReaderAttached: true,
+  hasChargerAttached: true,
+  hasLowBattery: false,
+  hasPrinterAttached: true,
+  machineConfig: { appMode: VxMarkOnly, machineId: '0000' },
+}
 
-  private initialUserState: Omit<State, keyof HardwareState> = {
-    ...this.initialVoterState,
-    ...this.initialCardPresentState,
-    ...this.sharedState,
-  }
+const initialSharedState: SharedState = {
+  appPrecinctId: '',
+  ballotsPrintedCount: 0,
+  electionDefinition: undefined,
+  isLiveMode: false,
+  isPollsOpen: false,
+  tally: [],
+}
 
-  private initialAppState: State = {
-    ...this.initialUserState,
-    ...this.initialHardwareState,
-  }
+const initialOtherState: OtherState = {
+  lastVoteUpdateAt: 0,
+  lastVoteSaveToCardAt: 0,
+  forceSaveVoteFlag: false,
+  writingVoteToCard: false,
+  lastCardDataString: '',
+}
 
-  public state = this.initialAppState
+const initialUserState: InitialUserState = {
+  ...initialVoterState,
+  ...initialCardState,
+  ...initialSharedState,
+}
 
-  public setMachineConfig = async (): Promise<void> => {
-    try {
-      const machineConfig = await this.props.machineConfig.get()
+const initialAppState: State = {
+  ...initialUserState,
+  ...initialHardwareState,
+  ...initialOtherState,
+}
 
-      if (machineConfig) {
-        this.setState({ machineConfig })
-      }
-    } catch {
-      // TODO: what should happen if `machineConfig` is not returned?
+// Move this function to another file and rewrite in FP?
```

Comment:
This comment doesn't give me enough context to know what to do. Also, add `TODO` prefix?